TripletCriterion.cu

```diff
@@ -7,21 +7,35 @@
 __global__ void triplet_dist_kernel(const int n, const float norm,
                                     const int nb_batch, const int length,
                                     float* x, float* y) {
+
+  // Unnecessary for loop...
+  // This is EXACTLY equivalent to `if (i < n)` statement
   for (int i = blockIdx.x*blockDim.x+threadIdx.x; i < n; i += blockDim.x*gridDim.x) {
```

Comment:
This is for GPU thread indexing and it is a essential part when the size of tasks exceeds the number of threads although it will not happen in practice.